edb/server/mng_port/edgecon.pyx

```diff
@@ -824,6 +868,10 @@ cdef class EdgeConnection:
             self.debug_print('SIMPLE QUERY', eql)
 
         if self.dbview.in_tx_error():
+            if not (allow_capabilities & enums.Capability.TRANSACTION):
+                raise errors.DisabledCapabilityError(
+                    f"Cannot recover transaction, disallowed capability"
```

Comment:
I suggest to move this check into `_recover_script_error`, where you know what was compiled, and where a correct error would be raised if the user didn't actually sent ROLLBACK or ROLLBACK TO SAVEPOINT. Then rephrase the error to

"Cannot execute ROLLBACK command; the transactions capability is disabled"

Reply:
Thanks. Fixed!